packages/validator/src/services/indices.ts

```diff
@@ -101,15 +125,37 @@ export class IndicesService {
     ApiError.assert(res, "Can not fetch state validators from beacon node");
 
     const newIndices = [];
+
+    const allValidatorStatuses = new MapDef<SimpleValidatorStatus, number>(() => 0);
+
     for (const validatorState of res.response.data) {
+      // Group all validators by status
+      const status = statusToSimpleStatusMapping(validatorState.status);
+      allValidatorStatuses.set(status, allValidatorStatuses.getOrDefault(status) + 1);
+
       const pubkeyHex = toHexString(validatorState.validator.pubkey);
       if (!this.pubkey2index.has(pubkeyHex)) {
-        this.logger.debug("Discovered validator", {pubkey: pubkeyHex, index: validatorState.index});
+        this.logger.debug("Validator exists in beacon chain", {
+          validatorIndex: validatorState.index,
+          pubkey: pubkeyHex,
+        });
         this.pubkey2index.set(pubkeyHex, validatorState.index);
         this.index2pubkey.set(validatorState.index, pubkeyHex);
         newIndices.push(validatorState.index);
       }
     }
+
+    // Retrieve the number of validators for each status
+    const statuses = Object.fromEntries(Array.from(allValidatorStatuses.entries()).filter((entry) => entry[1] > 0));
+    // The number of validators that are not in the beacon chain
+    const pendingCount = pubkeysHex.length - res.response.data.length;
+    if (pendingCount > 0) {
+      allValidatorStatuses.set("pending", allValidatorStatuses.getOrDefault("pending") + pendingCount);
+    }
```

Comment:
This needs to be before L149...
Also debatably remove the if (pendingCount > 0) condition

Reply:
Good point I think it can be removed since we our default function sets it to zero if it's undefined.